### ISSUE TITLE
Implement Principal Component Analysis

### ISSUE THREAD

**Issue by AtheMathmo** [AtheMathmo/rusty-machine#104]
[Rulinalg](https://github.com/AtheMathmo/rulinalg) now supports SVD - it would be nice to have PCA as a result.

**Comment by jprob** [comment]
I could do this, I am quite new to rust but I'd love to get involved here and PCA should be doable. Is there any good starting point into how the stuff for this project should look like?

**Comment by AtheMathmo** [comment]
There is actually an (in?)active PR right now for PCA ( #158 ). Maybe @sinhrks can say whether he will want to continue this or pass it on. I'd say that reading through the PR (and attached discussion) is a good starting point.

If you do end up taking this over I am happy to help out wherever I can. Sadly I am very busy right now but will do my best to find the time!

**Comment by AtheMathmo** [comment]
Yes, please feel free to suggest any other techniques! t-SNE would be a cool one to have added.

I think that dimension reduction is an algorithm that the current rusty-machine API doesn't capture well. I think the `Transformer` trait is probably the best way to implement one.

Let me know if you need any help finding/understanding things.

**Comment by tomtung** [comment]
Since there's SVD implemented, might also be interesting to introduce collatborative filtering methods into the library..?
